eZ/Publish/Core/MVC/Symfony/View/Builder/ContentViewBuilder.php

```diff
@@ -227,7 +234,11 @@ function (Repository $repository) use ($locationId) {
             }
         );
         if ($location->invisible) {
-            throw new HiddenLocationException($location, 'Location cannot be displayed as it is flagged as invisible.');
+            $request = $this->requestStack->getCurrentRequest();
+
+            if (!$request || !$request->attributes->get(PreviewController::PREVIEW_PARAMETER_NAME, false)) {
```

Comment:
Lack of Request object itself it is not reason to throw `HiddenLocationException`. We could check if `$request` is not null and then check value of attribute. Next we should throw exception.

Reply:
I don't think that's the case. The reason to throw `HiddenLocationException` is check done above:
`if ($location->invisible) {`
lack of `$request` is not a reason to throw an exception, but the whole check is to find a reason to withhold it.

Reply:
Ok, after reconsideration I can see your point!